### ISSUE TITLE
New images uploads without alt attribute should render an empty alt attribute

### ISSUE THREAD

**Issue by audrasjb** [WordPress/gutenberg#4359]
## Issue Overview

New images uploads without alt attribute should render an empty alt attribute.

## Steps to Reproduce (for bugs)

1. Add a simple image block in a new post.
2. Upload a new image to your website with this block.
3. Use it directly in your block (without alt text).
4. Open generated HTML code (both on back or front-end), you'll see the following HTML code:
```
<figure class="wp-block-image"><img src="my_image_path" /></figure>
``` 

## Expected Behavior
Even if empty, the alt attribute should always be present for every images. if this is not the case, some assistive technologies may vocalize the filename of the image or may have other types of harmful behavior.
A better generated HTML code should be:
```
<figure class="wp-block-image"><img src="my_image_path" alt="" /></figure>
``` 

## Current Behavior
ALT texts are empty for new images.

## Possible Solution
Fix image HTML rendering to use an empty alt attribute if no alt text in media metadata.
I can work on a workaround :)

## Todos
- [x] Tests
- [x] Patch
- [x] Tests
- [x] Documentation
  
Thanks @clairebizingre from WP-Francophone Slack team for showing this bug.

**Comment by audrasjb** [comment]
I'm currently testing a fix for this issue.
FYI, here is what I tested:
in `/blocks/library/image/block.js`, my diff is:
```
 	onSelectImage( media ) {
-		const attributes = { url: media.url, alt: media.alt, id: media.id };
+		const attributes = { url: media.url, id: media.id };
+		if ( media.alt ) {
+			attributes.alt = [ media.alt ];
+		} else {
+			attributes.alt = '';
+		}
 		if ( media.caption ) {
 			attributes.caption = [ media.caption ];
 		}
```

**Comment by audrasjb** [comment]
Ok sorry, that was stupid. We just need to set a default value on the block's attributes.
My PR is coming right now.

**Comment by afercia** [comment]
For reference, this is how screen readers read out an image without an alt attribute, when the image file name happens to be something totally meaningless 😬 

![screen shot 2018-01-11 at 09 15 57](https://user-images.githubusercontent.com/1682452/34818917-8bd84974-f6bc-11e7-8751-a41eb01508ec.jpg)

**Comment by audrasjb** [comment]
Hello @afercia thanks for your comments. 
This is exactly what I meaned when I wrote "Even if empty, the alt attribute should always be present for every images. if this is not the case, some assistive technologies may vocalize the filename of the image or may have other types of harmful behavior."
Thank you for your support on this issue/PR !

I guess this can be merged without risk, but I have an error with the fixtures tests and despite the Riad's help (in the WP-FR slack team) I can not regenerate them 😕  Halp! 😄

**Comment by afercia** [comment]
Same problem. I've tried
`GENERATE_MISSING_FIXTURES=y npm run test-unit -- --grep 'full post content fixture'`

but it doesn't work for me.

**Comment by schlessera** [comment]
In Drupal, the alt attribute is actually a required field, you cannot set an image without providing an alt attribute:

<img width="719" alt="image 2018-01-18 at 2 56 12 pm" src="https://user-images.githubusercontent.com/83631/35101411-be81cba4-fc5f-11e7-8970-c8a0d2722c2b.png">

Is this something worth considering?

**Comment by audrasjb** [comment]
Hi, thanks @schlessera but this is a bad practice. Alt attribute *must* be empty for decorative images for example.

**Comment by audrasjb** [comment]
(FYI this issue is going to be closed since my PR has been merged, see #4363 :)

**Comment by audrasjb** [comment]
Closed. See PR #4363
